idx/idx.go

```diff
@@ -168,4 +168,8 @@ type MetricIndex interface {
 	// If the fourth parameter is > 0 then only those metrics of which the LastUpdate
 	// time is >= the from timestamp will be included.
 	TagDetails(int, string, string, int64) (map[string]uint64, error)
+
+	// DeleteTagged deletes the specified series from the tag index and also the
+	// DefById index.
+	DeleteTagged(int, []string) ([]Archive, error)
```

Comment:
now that things are separated out, it's due time for us to document in `MemoryIdx` struct, which attributes are for what use case. it seems some properties are solely for tree index, some are for just tagged index, and some (just DefById ?) are for both. this can be documented/organized better.

Reply:
while doing that, i realized that `defByTagSet` needs to be keyed by `orgId`, so I did that, plus added comments to the properties of `MemoryIdx`: https://github.com/grafana/metrictank/pull/806/commits/f319903099d35645bed7899596289a22b2c2dc16

Reply:
I think you misunderstood what i meant.  I meant we need to comment the use case for each attribute (for tag index, for hierarchy index, or for both). there was no need to describe the internals of the structures again since you already described that where the structures are defined, which is the right thing to do.

internals of stuff should be documented where the stuff is defined, not where they are used.

I changed this via 20a4ba2797e1e1ffd9e41f98b08e9caa5d1d7748 please let me know what you think or if you see any mistakes

Reply:
yep, that works